### PR TITLE
Add Net Zero Tracker data

### PR DESCRIPTION
**Pull request by pabloarosado**
Add Net Zero Tracker data to ETL (it used to be fasttracked).
For reference, here is [the old dataset](https://owid.cloud/admin/datasets/6197), and [the resulting new dataset](http://staging-site-prosado/admin/datasets/6253) (on my staging server).
There's a mismatch of countries, as mentioned in https://github.com/owid/owid-issues/issues/705 , but that shouldn't affect this PR.

### CODE REVIEW

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  update_period_days: 365
+
+tables:
+  net_zero_tracker:
+    variables:
+      net_zero_status:
+        title: Status of net-zero carbon emissions targets
+        description_short: ""
+        description_from_producer: |-
+          More detail about each indicator is available in the Tracker Codebook: https://docs.google.com/document/d/13bGE8FCOeF5QweoOvyMPZi3XbZrGLEPL2aM04LuZQcc/edit?usp=sharing
```
Comment (paarriagadap) [inline etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml:26]: `[Tracker Codebook](https://docs.google.com/document/d/13bGE8FCOeF5QweoOvyMPZi3XbZrGLEPL2aM04LuZQcc/edit?usp=sharing)` ?

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    #  Bolivia had status "In policy document" but no year.
+    tb = tb.dropna(how="any").reset_index(drop=True)
+
+    # Harmonize country names.
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Add a column that simply indicates whether the country has a net zero target.
+    # NOTE: All countries in the table have set a net zero target. Those that have not are not in the table (and will
+    # show in charts as missing data).
+    tb["has_net_zero_target"] = "Net-zero achieved or pledged"
```
Comment (paarriagadap) [inline etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.py:46]: Don't we have the problem that we will be using "No data" as "Not pledged"?

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+dataset:
+  update_period_days: 365
+
+tables:
+  net_zero_tracker:
+    variables:
+      net_zero_status:
+        title: Status of net-zero carbon emissions targets
+        description_short: ""
```
Comment (paarriagadap) [inline etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml:24]: No `description_short`, when it's required

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+definitions:
```
Comment (paarriagadap) [inline etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml:1]: Is this a new configuration? I see it works for each indicator

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml (hunk 1 above)
Comment (paarriagadap) [reply etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml:26]: Isn't this redundant with the common definition above?

File: etl/steps/data/meadow/emissions/2023-10-10/net_zero_tracker.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    #
+    # Process data.
+    #
+    # Fix wrong column types.
+    for column in ["end_target_text", "interim_target_intensity_unit"]:
+        tb[column] = tb[column].fillna("").astype(str)
+
+    # Set an appropriate index and sort conveniently.
+    # NOTE: There are multiple rows for the same country-year (this will be handled in the garden step).
+    tb = tb.set_index(["country", "end_target_year"], verify_integrity=False).sort_index()
```
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/meadow/emissions/2023-10-10/net_zero_tracker.py:28]: Just to avoid the `UserWarning`

**Review by paarriagadap — APPROVED** [review]
Good! I added some minor comments

File: etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml (hunk 3 above)
Comment (pabloarosado) [reply etl/steps/data/garden/emissions/2023-10-10/net_zero_tracker.meta.yml:24]: Good point, I think `description_short` should complement the title. In this case, I think the title is descriptive enough. But I see that in the guidelines, this field is required. I'll discuss this with Joe.

**Comment by pabloarosado** [comment]
Thank you for the suggestions @paarriagadap!

**Pull request merged**